From 9a4cc3a7cc7e019c44624c5492e7a1a873d6a3da Mon Sep 17 00:00:00 2001
From: Xitij Ritesh Patel <xpatel@pulsecode.ca>
Date: Thu, 31 May 2012 00:54:04 -0400
Subject: [PATCH 1/2] Added QNX OS and removed doc generation

Modified CMakeLists.txt to add an option for QNX OS (not detected properly) and
to remove the generation of docs.
---
 CMakeLists.txt |   19 ++++++++++---------
 1 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7ffbd5..719d4c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ if(USE_XVFB)
     endif()
 endif()
 
-option(BUILD_TESTS "Build tests." TRUE)
+option(BUILD_TESTS "Build tests." FALSE)
 option(ENABLE_VERSION_SUFFIX "Used to use current version in suffix to generated files. This is used to allow multiples versions installed simultaneous." FALSE)
 set(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)" )
 set(LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE PATH "The subdirectory relative to the install prefix where libraries will be installed (default is /lib${LIB_SUFFIX})" FORCE)
@@ -105,7 +105,8 @@ elseif(Q_WS_SIMULATOR)
     set(ENABLE_SIMULATOR "1")
     set(AUTO_OS "simulator")
 else()
-    message(FATAL_ERROR "OS not supported")
+    set(AUTO_OS "qnx")
+#     message(FATAL_ERROR "OS not supported")
 endif()
 message(STATUS "Detected OS: ${AUTO_OS}")
 
@@ -140,7 +141,7 @@ add_custom_target(dist
 
 if (NOT SITE_PACKAGE)
     execute_process(
-        COMMAND ${SHIBOKEN_PYTHON_INTERPRETER} -c "from distutils import sysconfig; \\
+        COMMAND python3 -c "from distutils import sysconfig; \\
             print(sysconfig.get_python_lib(1,0,prefix='${CMAKE_INSTALL_PREFIX}'))"
         OUTPUT_VARIABLE SITE_PACKAGE
         OUTPUT_STRIP_TRAILING_WHITESPACE)
@@ -171,9 +172,9 @@ if (BUILD_TESTS)
     add_subdirectory(tests)
 endif ()
 
-find_program(DOT_EXEC dot)
-if (QT_SRC_DIR AND DOT_EXEC)
-    add_subdirectory(doc)
-else ()
-    message(STATUS "QT_SRC_DIR variable not set or graphviz not found, apidoc generation targets disabled.")
-endif()
+# find_program(DOT_EXEC dot)
+# if (QT_SRC_DIR AND DOT_EXEC)
+#     add_subdirectory(doc)
+# else ()
+#     message(STATUS "QT_SRC_DIR variable not set or graphviz not found, apidoc generation targets disabled.")
+# endif()
-- 
1.7.5.4


From 0902b974f847e732b4524df5d87368a85e5bfec7 Mon Sep 17 00:00:00 2001
From: Xitij Ritesh Patel <xpatel@pulsecode.ca>
Date: Thu, 31 May 2012 00:56:07 -0400
Subject: [PATCH 2/2] Modified typesystem to accomodate QNX OS

Modified the typesystem files to accomodate QNX OS build option, including
removal of QSound wrapper.
---
 PySide/QtCore/typesystem_core_qnx.xml  |   26 ++++++++++++++++++++++++++
 PySide/QtGui/CMakeLists.txt            |    1 -
 PySide/QtGui/typesystem_gui_common.xml |    2 +-
 PySide/QtGui/typesystem_gui_qnx.xml    |   28 ++++++++++++++++++++++++++++
 4 files changed, 55 insertions(+), 2 deletions(-)
 create mode 100644 PySide/QtCore/typesystem_core_qnx.xml
 create mode 100644 PySide/QtGui/typesystem_gui_qnx.xml

diff --git a/PySide/QtCore/typesystem_core_qnx.xml b/PySide/QtCore/typesystem_core_qnx.xml
new file mode 100644
index 0000000..eba223c
--- /dev/null
+++ b/PySide/QtCore/typesystem_core_qnx.xml
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<!--
+    This file is part of PySide project.
+    Copyright (C) 2011 Nokia Corporation and/or its subsidiary(-ies).
+    Contact: PySide team <contact@pyside.org>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+-->
+<typesystem package="PySide.QtCore">
+  <primitive-type name="Qt::HANDLE" target-lang-api-name="PyLong">
+    <!-- FIXME APIExtractor or shiboken do not support multiple includes by primitive type -->
+    <include file-name="QTextDocument" location="global"/>
+  </primitive-type>
+</typesystem>
diff --git a/PySide/QtGui/CMakeLists.txt b/PySide/QtGui/CMakeLists.txt
index 7625634..e506df1 100644
--- a/PySide/QtGui/CMakeLists.txt
+++ b/PySide/QtGui/CMakeLists.txt
@@ -275,7 +275,6 @@ ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qshowevent_wrapper.cpp
 ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qsizepolicy_wrapper.cpp
 ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qslider_wrapper.cpp
 ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qsortfilterproxymodel_wrapper.cpp
-${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qsound_wrapper.cpp
 ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qspaceritem_wrapper.cpp
 ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qspinbox_wrapper.cpp
 ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qsplashscreen_wrapper.cpp
diff --git a/PySide/QtGui/typesystem_gui_common.xml b/PySide/QtGui/typesystem_gui_common.xml
index ab97ee1..90d87e6 100644
--- a/PySide/QtGui/typesystem_gui_common.xml
+++ b/PySide/QtGui/typesystem_gui_common.xml
@@ -2584,7 +2584,7 @@
       </inject-code>
     </add-function>
   </object-type>
-  <object-type name="QSound"/>
+  <!--object-type name="QSound"/-->
   <object-type name="QSpacerItem"/>
   <object-type name="QStandardItem">
     <enum-type name="ItemType"/>
diff --git a/PySide/QtGui/typesystem_gui_qnx.xml b/PySide/QtGui/typesystem_gui_qnx.xml
new file mode 100644
index 0000000..a7b1f33
--- /dev/null
+++ b/PySide/QtGui/typesystem_gui_qnx.xml
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<!--
+    This file is part of PySide project.
+    Copyright (C) 2009-2010 Nokia Corporation and/or its subsidiary(-ies).
+    Contact: PySide team <contact@pyside.org>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+-->
+<typesystem package="PySide.QtGui">
+  <!-- Qt::HANDLE is typedef to "void *", which gives compilation errors on
+       conversion templates -->
+  <value-type name="QCursor">
+    <modify-function signature="QCursor(Qt::HANDLE)" remove="all"/>
+  </value-type>
+  <rejection class="QSound"/>
+</typesystem>
-- 
1.7.5.4

